src/main/java/org/folio/oaipmh/mappers/AbstractMapper.java

```diff
@@ -7,24 +14,12 @@
 import org.marc4j.marc.Record;
 import org.w3c.dom.Node;
 
-import javax.xml.transform.dom.DOMResult;
-import java.io.*;
-import java.nio.charset.StandardCharsets;
 
-/**
- * Converts MarkJson format to MarcXML format.
- */
-public class MarcXmlMapper implements Mapper {
+public abstract class AbstractMapper implements Mapper {
 
-  /**
-   * {@inheritDoc}
-   * @param source String representation of MarkJson source.
-   * @return the Node object that represents result MarcXML document
-   */
-  @Override
-  public Node convert(String source) {
-    try(InputStream inputStream
-          = new ByteArrayInputStream(source.getBytes(StandardCharsets.UTF_8))) {
+  protected Node templateConvert(String source) {
```

Comment:
Instead of providing 'helper' templateConvert method, I would propose to provide default implementation of covert method that:
- does what templateConvert currently does
- and then call a kind of postProcess method which is empty by default, but can be overridden by children, e.g. XSLT transformer to perform XSLT transformation